nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/UnpackContent.java

```diff
@@ -406,50 +399,118 @@ public ZipUnpacker(final Pattern fileFilter, final char[] password) {
         @Override
         public void unpack(final ProcessSession session, final FlowFile source, final List<FlowFile> unpacked) {
             final String fragmentId = UUID.randomUUID().toString();
-            session.read(source, new InputStreamCallback() {
-                @Override
-                public void process(final InputStream in) throws IOException {
-                    int fragmentCount = 0;
-                    try (final ZipInputStream zipIn = new ZipInputStream(new BufferedInputStream(in), password)) {
-                        LocalFileHeader zipEntry;
-                        while ((zipEntry = zipIn.getNextEntry()) != null) {
-                            final String zipEntryName = zipEntry.getFileName();
-                            if (zipEntry.isDirectory() || !fileMatches(zipEntryName)) {
-                                continue;
-                            }
-                            final File file = new File(zipEntryName);
-                            final String parentDirectory = (file.getParent() == null) ? "/" : file.getParent();
-                            final Path absPath = file.toPath().toAbsolutePath();
-                            final String absPathString = absPath.getParent().toString() + "/";
-
-                            FlowFile unpackedFile = session.create(source);
-                            try {
-                                final Map<String, String> attributes = new HashMap<>();
-                                attributes.put(CoreAttributes.FILENAME.key(), file.getName());
-                                attributes.put(CoreAttributes.PATH.key(), parentDirectory);
-                                attributes.put(CoreAttributes.ABSOLUTE_PATH.key(), absPathString);
-                                attributes.put(CoreAttributes.MIME_TYPE.key(), OCTET_STREAM);
+            if (password == null) {
```

Comment:
Is there a good place in the docs to mention this? Maybe the description of the password property? I realize it's under the hood but seems like we should mention it in case of regressions, unexpected/changed behavior, etc.?

Reply:
What do you think about adding something to the Password property description?  NiFi 1.13.0 introduced support for the Password property and changed the library from Apache Commons Compress to Zip4j.  The changes in this PR restore the default behavior to use Apache Commons Compress and only use Zip4j when a Password is configured.

Zip compression algorithm support depends on the version of Apache Commons Compress is not something explicitly advertised, so alternative compression algorithms are an edge case, but common enough.

Reply:
That's what I was thinking too. Most people might not care about such details, but it would be there if you need it :)

Reply:
Sounds good, I added a note to the Password property description indicating that alternative Zip compression algorithms will be disabled when configuring a password.

Reply:
By "disabled", does that mean the user can set an invalid combination that would only be determined at runtime? If so maybe we could add a customValidate() or something that checks the algorithm and the password to see if they are a valid combination

Reply:
No, it is not possible to configure the supported Zip compression algorithms.  Support is implicit based on the library used.  With being the case, it is not possible to set an invalid combination of settings, so custom validation should not be necessary.